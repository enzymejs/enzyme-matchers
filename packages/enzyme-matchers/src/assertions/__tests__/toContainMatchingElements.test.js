const PropTypes = require('prop-types');
const getNodeName = require('../../utils/name');
const toContainMatchingElements = require('../toContainMatchingElements');

function User(props) {
  return (
    <span className={props.className}>
      User <span data-index={`value-${props.index}`}>{props.index}</span>
    </span>
  );
}

User.propTypes = {
  index: PropTypes.number.isRequired,
  className: PropTypes.string,
};

function Fixture() {
  return (
    <div>
      <ul>
        <li>
          <User index={1} className="userOne" />
        </li>
        <li>
          <User index={2} className="userTwo" />
        </li>
      </ul>
    </div>
  );
}

describe('toContainMatchingElements', () => {
  [shallow, mount].forEach(renderer => {
    describe(`with a ${renderer.name} wrapper`, () => {
      const wrapper = renderer(<Fixture />);
      const argsToTest = [
        ['2 User', wrapper, 2, 'User', true],
        ['1 [index=1]', wrapper, 1, '[index=1]', true],
        ['2 [index]', wrapper, 2, '[index]', true],
        ['2 [index] in ul', wrapper.find('ul'), 2, '[index]', true],
        ['1 [index=1] in User', wrapper.find('User'), 1, '[index=1]', true],
        ['3 User', wrapper, 3, 'User', false],
        ['1 .userThree', wrapper, 1, '.userThree', false],
        ['1 [index]', wrapper, 1, '[index]', false],
        ['3 [index] in ul', wrapper.find('ul'), 3, '[index]', false],
        ['1 [data-index]', wrapper, 1, '[data-index]', false],
        ...(renderer === shallow
          ? [
              ['1 .userOne', wrapper, 1, '.userOne', true],
              ['0 [data-index]', wrapper, 0, '[data-index]', true],
              [
                '1 [data-index="value-1"]',
                wrapper,
                1,
                '[data-index="value-1"]',
                false,
              ],
            ]
          : []),
        ...(renderer === mount
          ? [
              ['2 .userOne', wrapper, 2, '.userOne', true],
              ['2 [data-index]', wrapper, 2, '[data-index]', true],
              [
                '1 [data-index="value-1"]',
                wrapper,
                1,
                '[data-index="value-1"]',
                true,
              ],
            ]
          : []),
      ];

      argsToTest.forEach(
        ([description, currentWrapper, count, selector, expectedResult]) => {
          const result = toContainMatchingElements(
            currentWrapper,
            count,
            selector
          );
          it(`returns ${expectedResult} for "${selector}" in <${getNodeName(
            currentWrapper
          )}>`, () => {
            expect(result.pass).toBe(expectedResult);
          });

          it(`returns the proper verbage for "${description}"`, () => {
            if (expectedResult) {
              expect(result.message).toMatchSnapshot();
            } else {
              expect(result.negatedMessage).toMatchSnapshot();
            }
          });

          it('provides contextual information for the message', () => {
            expect(result.contextualInformation).toMatchSnapshot();
          });
        }
      );
    });
  });
});
