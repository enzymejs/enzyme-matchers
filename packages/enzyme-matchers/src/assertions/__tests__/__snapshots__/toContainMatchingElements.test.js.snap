// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`toContainMatchingElements with a mount wrapper provides contextual information for the message 1`] = `
Object {
  "actual": "Element tree for <Fixture>:
 <Fixture>
  <div>
    <ul>
      <li>
        <User index={1} className=\\"userOne\\">
          <span className=\\"userOne\\">
            User 
            <span data-index=\\"value-1\\">
              1
            </span>
          </span>
        </User>
      </li>
      <li>
        <User index={2} className=\\"userTwo\\">
          <span className=\\"userTwo\\">
            User 
            <span data-index=\\"value-2\\">
              2
            </span>
          </span>
        </User>
      </li>
    </ul>
  </div>
</Fixture>",
}
`;

exports[`toContainMatchingElements with a mount wrapper provides contextual information for the message 2`] = `
Object {
  "actual": "Element tree for <Fixture>:
 <Fixture>
  <div>
    <ul>
      <li>
        <User index={1} className=\\"userOne\\">
          <span className=\\"userOne\\">
            User 
            <span data-index=\\"value-1\\">
              1
            </span>
          </span>
        </User>
      </li>
      <li>
        <User index={2} className=\\"userTwo\\">
          <span className=\\"userTwo\\">
            User 
            <span data-index=\\"value-2\\">
              2
            </span>
          </span>
        </User>
      </li>
    </ul>
  </div>
</Fixture>",
}
`;

exports[`toContainMatchingElements with a mount wrapper provides contextual information for the message 3`] = `
Object {
  "actual": "Element tree for <Fixture>:
 <Fixture>
  <div>
    <ul>
      <li>
        <User index={1} className=\\"userOne\\">
          <span className=\\"userOne\\">
            User 
            <span data-index=\\"value-1\\">
              1
            </span>
          </span>
        </User>
      </li>
      <li>
        <User index={2} className=\\"userTwo\\">
          <span className=\\"userTwo\\">
            User 
            <span data-index=\\"value-2\\">
              2
            </span>
          </span>
        </User>
      </li>
    </ul>
  </div>
</Fixture>",
}
`;

exports[`toContainMatchingElements with a mount wrapper provides contextual information for the message 4`] = `
Object {
  "actual": "Element tree for <ul>:
 <ul>
  <li>
    <User index={1} className=\\"userOne\\">
      <span className=\\"userOne\\">
        User 
        <span data-index=\\"value-1\\">
          1
        </span>
      </span>
    </User>
  </li>
  <li>
    <User index={2} className=\\"userTwo\\">
      <span className=\\"userTwo\\">
        User 
        <span data-index=\\"value-2\\">
          2
        </span>
      </span>
    </User>
  </li>
</ul>",
}
`;

exports[`toContainMatchingElements with a mount wrapper provides contextual information for the message 5`] = `
Object {
  "actual": "Element tree for <2 (anonymous) nodes found>:
 Multiple nodes found:
0: <function User(props) {
  return React.createElement(\\"span\\", {
    className: props.className
  }, \\"User \\", React.createElement(\\"span\\", {
    \\"data-index\\": \`value-\${props.index}\`
  }, props.index));
} index=\\"1\\" className=\\"userOne\\"/>
1: <function User(props) {
  return React.createElement(\\"span\\", {
    className: props.className
  }, \\"User \\", React.createElement(\\"span\\", {
    \\"data-index\\": \`value-\${props.index}\`
  }, props.index));
} index=\\"2\\" className=\\"userTwo\\"/>
",
}
`;

exports[`toContainMatchingElements with a mount wrapper provides contextual information for the message 6`] = `
Object {
  "actual": "Element tree for <Fixture>:
 <Fixture>
  <div>
    <ul>
      <li>
        <User index={1} className=\\"userOne\\">
          <span className=\\"userOne\\">
            User 
            <span data-index=\\"value-1\\">
              1
            </span>
          </span>
        </User>
      </li>
      <li>
        <User index={2} className=\\"userTwo\\">
          <span className=\\"userTwo\\">
            User 
            <span data-index=\\"value-2\\">
              2
            </span>
          </span>
        </User>
      </li>
    </ul>
  </div>
</Fixture>",
}
`;

exports[`toContainMatchingElements with a mount wrapper provides contextual information for the message 7`] = `
Object {
  "actual": "Element tree for <Fixture>:
 <Fixture>
  <div>
    <ul>
      <li>
        <User index={1} className=\\"userOne\\">
          <span className=\\"userOne\\">
            User 
            <span data-index=\\"value-1\\">
              1
            </span>
          </span>
        </User>
      </li>
      <li>
        <User index={2} className=\\"userTwo\\">
          <span className=\\"userTwo\\">
            User 
            <span data-index=\\"value-2\\">
              2
            </span>
          </span>
        </User>
      </li>
    </ul>
  </div>
</Fixture>",
}
`;

exports[`toContainMatchingElements with a mount wrapper provides contextual information for the message 8`] = `
Object {
  "actual": "Element tree for <Fixture>:
 <Fixture>
  <div>
    <ul>
      <li>
        <User index={1} className=\\"userOne\\">
          <span className=\\"userOne\\">
            User 
            <span data-index=\\"value-1\\">
              1
            </span>
          </span>
        </User>
      </li>
      <li>
        <User index={2} className=\\"userTwo\\">
          <span className=\\"userTwo\\">
            User 
            <span data-index=\\"value-2\\">
              2
            </span>
          </span>
        </User>
      </li>
    </ul>
  </div>
</Fixture>",
}
`;

exports[`toContainMatchingElements with a mount wrapper provides contextual information for the message 9`] = `
Object {
  "actual": "Element tree for <ul>:
 <ul>
  <li>
    <User index={1} className=\\"userOne\\">
      <span className=\\"userOne\\">
        User 
        <span data-index=\\"value-1\\">
          1
        </span>
      </span>
    </User>
  </li>
  <li>
    <User index={2} className=\\"userTwo\\">
      <span className=\\"userTwo\\">
        User 
        <span data-index=\\"value-2\\">
          2
        </span>
      </span>
    </User>
  </li>
</ul>",
}
`;

exports[`toContainMatchingElements with a mount wrapper provides contextual information for the message 10`] = `
Object {
  "actual": "Element tree for <Fixture>:
 <Fixture>
  <div>
    <ul>
      <li>
        <User index={1} className=\\"userOne\\">
          <span className=\\"userOne\\">
            User 
            <span data-index=\\"value-1\\">
              1
            </span>
          </span>
        </User>
      </li>
      <li>
        <User index={2} className=\\"userTwo\\">
          <span className=\\"userTwo\\">
            User 
            <span data-index=\\"value-2\\">
              2
            </span>
          </span>
        </User>
      </li>
    </ul>
  </div>
</Fixture>",
}
`;

exports[`toContainMatchingElements with a mount wrapper provides contextual information for the message 11`] = `
Object {
  "actual": "Element tree for <Fixture>:
 <Fixture>
  <div>
    <ul>
      <li>
        <User index={1} className=\\"userOne\\">
          <span className=\\"userOne\\">
            User 
            <span data-index=\\"value-1\\">
              1
            </span>
          </span>
        </User>
      </li>
      <li>
        <User index={2} className=\\"userTwo\\">
          <span className=\\"userTwo\\">
            User 
            <span data-index=\\"value-2\\">
              2
            </span>
          </span>
        </User>
      </li>
    </ul>
  </div>
</Fixture>",
}
`;

exports[`toContainMatchingElements with a mount wrapper provides contextual information for the message 12`] = `
Object {
  "actual": "Element tree for <Fixture>:
 <Fixture>
  <div>
    <ul>
      <li>
        <User index={1} className=\\"userOne\\">
          <span className=\\"userOne\\">
            User 
            <span data-index=\\"value-1\\">
              1
            </span>
          </span>
        </User>
      </li>
      <li>
        <User index={2} className=\\"userTwo\\">
          <span className=\\"userTwo\\">
            User 
            <span data-index=\\"value-2\\">
              2
            </span>
          </span>
        </User>
      </li>
    </ul>
  </div>
</Fixture>",
}
`;

exports[`toContainMatchingElements with a mount wrapper provides contextual information for the message 13`] = `
Object {
  "actual": "Element tree for <Fixture>:
 <Fixture>
  <div>
    <ul>
      <li>
        <User index={1} className=\\"userOne\\">
          <span className=\\"userOne\\">
            User 
            <span data-index=\\"value-1\\">
              1
            </span>
          </span>
        </User>
      </li>
      <li>
        <User index={2} className=\\"userTwo\\">
          <span className=\\"userTwo\\">
            User 
            <span data-index=\\"value-2\\">
              2
            </span>
          </span>
        </User>
      </li>
    </ul>
  </div>
</Fixture>",
}
`;

exports[`toContainMatchingElements with a mount wrapper returns the proper verbage for "1 .userThree" 1`] = `"Expected <Fixture> to not contain 1 element matching \\".userThree\\" but it did."`;

exports[`toContainMatchingElements with a mount wrapper returns the proper verbage for "1 [data-index="value-1"]" 1`] = `"Expected <Fixture> to contain 1 element matching \\"[data-index=\\"value-1\\"]\\" but 1 was found."`;

exports[`toContainMatchingElements with a mount wrapper returns the proper verbage for "1 [data-index]" 1`] = `"Expected <Fixture> to not contain 1 element matching \\"[data-index]\\" but it did."`;

exports[`toContainMatchingElements with a mount wrapper returns the proper verbage for "1 [index=1] in User" 1`] = `"Expected <2 (anonymous) nodes found> to contain 1 element matching \\"[index=1]\\" but 1 was found."`;

exports[`toContainMatchingElements with a mount wrapper returns the proper verbage for "1 [index=1]" 1`] = `"Expected <Fixture> to contain 1 element matching \\"[index=1]\\" but 1 was found."`;

exports[`toContainMatchingElements with a mount wrapper returns the proper verbage for "1 [index]" 1`] = `"Expected <Fixture> to not contain 1 element matching \\"[index]\\" but it did."`;

exports[`toContainMatchingElements with a mount wrapper returns the proper verbage for "2 .userOne" 1`] = `"Expected <Fixture> to contain 2 elements matching \\".userOne\\" but 2 were found."`;

exports[`toContainMatchingElements with a mount wrapper returns the proper verbage for "2 [data-index]" 1`] = `"Expected <Fixture> to contain 2 elements matching \\"[data-index]\\" but 2 were found."`;

exports[`toContainMatchingElements with a mount wrapper returns the proper verbage for "2 [index] in ul" 1`] = `"Expected <ul> to contain 2 elements matching \\"[index]\\" but 2 were found."`;

exports[`toContainMatchingElements with a mount wrapper returns the proper verbage for "2 [index]" 1`] = `"Expected <Fixture> to contain 2 elements matching \\"[index]\\" but 2 were found."`;

exports[`toContainMatchingElements with a mount wrapper returns the proper verbage for "2 User" 1`] = `"Expected <Fixture> to contain 2 elements matching \\"User\\" but 2 were found."`;

exports[`toContainMatchingElements with a mount wrapper returns the proper verbage for "3 [index] in ul" 1`] = `"Expected <ul> to not contain 3 elements matching \\"[index]\\" but it did."`;

exports[`toContainMatchingElements with a mount wrapper returns the proper verbage for "3 User" 1`] = `"Expected <Fixture> to not contain 3 elements matching \\"User\\" but it did."`;

exports[`toContainMatchingElements with a shallow wrapper provides contextual information for the message 1`] = `
Object {
  "actual": "Element tree for <div>:
 <div>
  <ul>
    <li>
      <User index={1} className=\\"userOne\\" />
    </li>
    <li>
      <User index={2} className=\\"userTwo\\" />
    </li>
  </ul>
</div>",
}
`;

exports[`toContainMatchingElements with a shallow wrapper provides contextual information for the message 2`] = `
Object {
  "actual": "Element tree for <div>:
 <div>
  <ul>
    <li>
      <User index={1} className=\\"userOne\\" />
    </li>
    <li>
      <User index={2} className=\\"userTwo\\" />
    </li>
  </ul>
</div>",
}
`;

exports[`toContainMatchingElements with a shallow wrapper provides contextual information for the message 3`] = `
Object {
  "actual": "Element tree for <div>:
 <div>
  <ul>
    <li>
      <User index={1} className=\\"userOne\\" />
    </li>
    <li>
      <User index={2} className=\\"userTwo\\" />
    </li>
  </ul>
</div>",
}
`;

exports[`toContainMatchingElements with a shallow wrapper provides contextual information for the message 4`] = `
Object {
  "actual": "Element tree for <ul>:
 <ul>
  <li>
    <User index={1} className=\\"userOne\\" />
  </li>
  <li>
    <User index={2} className=\\"userTwo\\" />
  </li>
</ul>",
}
`;

exports[`toContainMatchingElements with a shallow wrapper provides contextual information for the message 5`] = `
Object {
  "actual": "Element tree for <2 (anonymous) nodes found>:
 Multiple nodes found:
0: <function User(props) {
  return React.createElement(\\"span\\", {
    className: props.className
  }, \\"User \\", React.createElement(\\"span\\", {
    \\"data-index\\": \`value-\${props.index}\`
  }, props.index));
} index=\\"1\\" className=\\"userOne\\"/>
1: <function User(props) {
  return React.createElement(\\"span\\", {
    className: props.className
  }, \\"User \\", React.createElement(\\"span\\", {
    \\"data-index\\": \`value-\${props.index}\`
  }, props.index));
} index=\\"2\\" className=\\"userTwo\\"/>
",
}
`;

exports[`toContainMatchingElements with a shallow wrapper provides contextual information for the message 6`] = `
Object {
  "actual": "Element tree for <div>:
 <div>
  <ul>
    <li>
      <User index={1} className=\\"userOne\\" />
    </li>
    <li>
      <User index={2} className=\\"userTwo\\" />
    </li>
  </ul>
</div>",
}
`;

exports[`toContainMatchingElements with a shallow wrapper provides contextual information for the message 7`] = `
Object {
  "actual": "Element tree for <div>:
 <div>
  <ul>
    <li>
      <User index={1} className=\\"userOne\\" />
    </li>
    <li>
      <User index={2} className=\\"userTwo\\" />
    </li>
  </ul>
</div>",
}
`;

exports[`toContainMatchingElements with a shallow wrapper provides contextual information for the message 8`] = `
Object {
  "actual": "Element tree for <div>:
 <div>
  <ul>
    <li>
      <User index={1} className=\\"userOne\\" />
    </li>
    <li>
      <User index={2} className=\\"userTwo\\" />
    </li>
  </ul>
</div>",
}
`;

exports[`toContainMatchingElements with a shallow wrapper provides contextual information for the message 9`] = `
Object {
  "actual": "Element tree for <ul>:
 <ul>
  <li>
    <User index={1} className=\\"userOne\\" />
  </li>
  <li>
    <User index={2} className=\\"userTwo\\" />
  </li>
</ul>",
}
`;

exports[`toContainMatchingElements with a shallow wrapper provides contextual information for the message 10`] = `
Object {
  "actual": "Element tree for <div>:
 <div>
  <ul>
    <li>
      <User index={1} className=\\"userOne\\" />
    </li>
    <li>
      <User index={2} className=\\"userTwo\\" />
    </li>
  </ul>
</div>",
}
`;

exports[`toContainMatchingElements with a shallow wrapper provides contextual information for the message 11`] = `
Object {
  "actual": "Element tree for <div>:
 <div>
  <ul>
    <li>
      <User index={1} className=\\"userOne\\" />
    </li>
    <li>
      <User index={2} className=\\"userTwo\\" />
    </li>
  </ul>
</div>",
}
`;

exports[`toContainMatchingElements with a shallow wrapper provides contextual information for the message 12`] = `
Object {
  "actual": "Element tree for <div>:
 <div>
  <ul>
    <li>
      <User index={1} className=\\"userOne\\" />
    </li>
    <li>
      <User index={2} className=\\"userTwo\\" />
    </li>
  </ul>
</div>",
}
`;

exports[`toContainMatchingElements with a shallow wrapper provides contextual information for the message 13`] = `
Object {
  "actual": "Element tree for <div>:
 <div>
  <ul>
    <li>
      <User index={1} className=\\"userOne\\" />
    </li>
    <li>
      <User index={2} className=\\"userTwo\\" />
    </li>
  </ul>
</div>",
}
`;

exports[`toContainMatchingElements with a shallow wrapper returns the proper verbage for "0 [data-index]" 1`] = `"Expected <div> to contain 0 elements matching \\"[data-index]\\" but 0 were found."`;

exports[`toContainMatchingElements with a shallow wrapper returns the proper verbage for "1 .userOne" 1`] = `"Expected <div> to contain 1 element matching \\".userOne\\" but 1 was found."`;

exports[`toContainMatchingElements with a shallow wrapper returns the proper verbage for "1 .userThree" 1`] = `"Expected <div> to not contain 1 element matching \\".userThree\\" but it did."`;

exports[`toContainMatchingElements with a shallow wrapper returns the proper verbage for "1 [data-index="value-1"]" 1`] = `"Expected <div> to not contain 1 element matching \\"[data-index=\\"value-1\\"]\\" but it did."`;

exports[`toContainMatchingElements with a shallow wrapper returns the proper verbage for "1 [data-index]" 1`] = `"Expected <div> to not contain 1 element matching \\"[data-index]\\" but it did."`;

exports[`toContainMatchingElements with a shallow wrapper returns the proper verbage for "1 [index=1] in User" 1`] = `"Expected <2 (anonymous) nodes found> to contain 1 element matching \\"[index=1]\\" but 1 was found."`;

exports[`toContainMatchingElements with a shallow wrapper returns the proper verbage for "1 [index=1]" 1`] = `"Expected <div> to contain 1 element matching \\"[index=1]\\" but 1 was found."`;

exports[`toContainMatchingElements with a shallow wrapper returns the proper verbage for "1 [index]" 1`] = `"Expected <div> to not contain 1 element matching \\"[index]\\" but it did."`;

exports[`toContainMatchingElements with a shallow wrapper returns the proper verbage for "2 [index] in ul" 1`] = `"Expected <ul> to contain 2 elements matching \\"[index]\\" but 2 were found."`;

exports[`toContainMatchingElements with a shallow wrapper returns the proper verbage for "2 [index]" 1`] = `"Expected <div> to contain 2 elements matching \\"[index]\\" but 2 were found."`;

exports[`toContainMatchingElements with a shallow wrapper returns the proper verbage for "2 User" 1`] = `"Expected <div> to contain 2 elements matching \\"User\\" but 2 were found."`;

exports[`toContainMatchingElements with a shallow wrapper returns the proper verbage for "3 [index] in ul" 1`] = `"Expected <ul> to not contain 3 elements matching \\"[index]\\" but it did."`;

exports[`toContainMatchingElements with a shallow wrapper returns the proper verbage for "3 User" 1`] = `"Expected <div> to not contain 3 elements matching \\"User\\" but it did."`;
